### PR TITLE
Fetch taint vulnerabilities

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Fixes https://github.com/SonarSource/sonarlint-visualstudio/issues/1990

### CODE REVIEW

File: SonarQube.Client/Api/V7_20/GetIssuesRequest.cs [hunk 1]
```diff
@@ -80,11 +80,11 @@ protected override SonarQubeIssue[] ParseResponse(string response)
         }
 
         private SonarQubeIssue ToSonarQubeIssue(ServerIssue issue) =>
-            new SonarQubeIssue(FilePathNormalizer.NormalizeSonarQubePath(ComputePath(issue.Component)), issue.Hash, issue.Line, issue.Message, ComputeModuleKey(issue),
+            new SonarQubeIssue(ComputePath(issue.Component), issue.Hash, issue.Line, issue.Message, ComputeModuleKey(issue),
                 GetRuleKey(issue.CompositeRuleKey), issue.Status == "RESOLVED", ToIssueFlows(issue.Flows));
 
         private string ComputePath(string component) =>
-            componentKeyPathLookup[component].FirstOrDefault() ?? string.Empty;
+            FilePathNormalizer.NormalizeSonarQubePath(componentKeyPathLookup[component].FirstOrDefault() ?? string.Empty);
```
Comment (rita-g-sonarsource) [inline SonarQube.Client/Api/V7_20/GetIssuesRequest.cs:87]: I missed this when I added FilePathNormalizer but flows were never used up until now so no harm was done.

File: SonarQube.Client.Tests/SonarQubeService_GetTaintVulnerabilitiesRequest.cs [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+            // should return only 2 taint out of 4 vulnerabilities
+            result.Count.Should().Be(2);
+
+            var taint1 = result[0];
+
+            taint1.RuleId.Should().Be("javasecurity:S2076");
+            taint1.Message.Should().Be("Refactor this code to not construct the OS command from tainted, user-controlled data.");
+            taint1.FilePath.Should().Be("src\\main\\java\\foo\\security\\injection\\CommandInjectionVulnerability.java");
+            taint1.Hash.Should().Be("1703916771e6abb765843e62a76fcb5a");
+            taint1.Line.Should().Be(17);
```
Comment (rita-g-sonarsource) [inline SonarQube.Client.Tests/SonarQubeService_GetTaintVulnerabilitiesRequest.cs:619]: FYI the API also returns TextRange field but we don't currently store it. The sink is in the locations as well so it's probably not that critical.

File: SonarQube.Client/Api/DefaultConfiguration.cs [hunk 3]
```diff
@@ -50,7 +50,8 @@ public static RequestFactory Configure(RequestFactory requestFactory)
                 .RegisterRequest<IGetRoslynExportProfileRequest, V6_60.GetRoslynExportProfileRequest>("6.6")
                 .RegisterRequest<IGetOrganizationsRequest, V7_00.GetOrganizationsRequest>("7.0")
                 .RegisterRequest<IGetIssuesRequest, V7_20.GetIssuesRequestWrapper>("7.2")
-                .RegisterRequest<IGetHotspotRequest, V8_6.GetHotspotRequest>("8.6");
+                .RegisterRequest<IGetHotspotRequest, V8_6.GetHotspotRequest>("8.6")
+                .RegisterRequest<IGetTaintVulnerabilitiesRequest, V7_20.GetTaintVulnerabilitiesRequest>("7.2");
```
Comment (duncanp-sonar) [inline SonarQube.Client/Api/DefaultConfiguration.cs:54]: Won't this mean that the "view taint vulnerabilities" feature will work against older versions of SonarQube from v7.2 onwards, and if so is it what we want?

From a technical point of view it probably doesn't make any difference, but it makes this more complicated from a marketing and support point of view. Probably worth checking what the PMs view is.

File: SonarQube.Client/Api/V7_20/GetTaintVulnerabilitiesRequest.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            getIssuesRequest.Logger = Logger;
+            getIssuesRequest.ProjectKey = ProjectKey;
+            getIssuesRequest.Statuses = "OPEN,CONFIRMED,REOPENED";
+            getIssuesRequest.Types = "VULNERABILITY";
+
+            var vulnerabilities = await getIssuesRequest.InvokeAsync(httpClient, token);
+            WarnForApiLimit(vulnerabilities, getIssuesRequest);
+
+            var taintVulnerabilities = vulnerabilities
+                .Where(x => x.RuleId.Contains("security"))
```
Comment (duncanp-sonar) [inline SonarQube.Client/Api/V7_20/GetTaintVulnerabilitiesRequest.cs:48]: Wasn't there a `repo` field that contained just contained the repo key? That would make it clearer what the intent of the code was.

**Review by duncanp-sonar — APPROVED** [review]

File: SonarQube.Client/Api/DefaultConfiguration.cs (hunk 3 above)
Comment (rita-g-sonarsource) [reply SonarQube.Client/Api/DefaultConfiguration.cs:54]: Good point, I asked Marco. For now he suggests 7.2, he will let us know if this changes.

File: SonarQube.Client/Api/V7_20/GetTaintVulnerabilitiesRequest.cs (hunk 4 above)
Comment (rita-g-sonarsource) [reply SonarQube.Client/Api/V7_20/GetTaintVulnerabilitiesRequest.cs:48]: No I don't see it in the response, and neither in the documentation: https://next.sonarqube.com/sonarqube/web_api/api/issues?query=issues#

**Pull request merged**